enterprise/backup/src/main/java/org/neo4j/backup/BackupCommandArgumentHandler.java

```diff
@@ -0,0 +1,150 @@
+import org.neo4j.commandline.arguments.MandatoryNamedArg;
+import org.neo4j.commandline.arguments.OptionalBooleanArg;
+import org.neo4j.commandline.arguments.OptionalNamedArg;
+import org.neo4j.commandline.arguments.common.MandatoryCanonicalPath;
+import org.neo4j.commandline.arguments.common.OptionalCanonicalPath;
+import org.neo4j.consistency.ConsistencyCheckSettings;
+import org.neo4j.consistency.checking.full.CheckConsistencyConfig;
+import org.neo4j.helpers.OptionalHostnamePort;
+import org.neo4j.helpers.TimeUtil;
+import org.neo4j.kernel.configuration.Config;
+
+import static org.neo4j.kernel.impl.util.Converters.toOptionalHostnamePortFromRawAddress;
+
+public class BackupCommandArgumentHandler
+{
+
+    private static final Arguments arguments = new Arguments()
+            .withArgument( new MandatoryCanonicalPath( "backup-dir", "backup-path",
+                    "Directory to place backup in." ) )
+            .withArgument( new MandatoryNamedArg( "name", "graph.db-backup",
+                    "Name of backup. If a backup with this name already exists an incremental backup will be " +
+                            "attempted." ) )
+            .withArgument( new OptionalNamedArg( "from", "address", "localhost:6362",
+                    "Host and port of Neo4j." ) )
+            .withArgument( new OptionalBooleanArg( "fallback-to-full", true,
+                    "If an incremental backup fails backup will move the old backup to <name>.err.<N> and " +
+                            "fallback to a full backup instead." ) )
+            .withArgument( new OptionalNamedArg( "timeout", "timeout", "20m",
+                    "Timeout in the form <time>[ms|s|m|h], where the default unit is seconds." ) )
+            .withArgument( new OptionalBooleanArg( "check-consistency", true,
+                    "If a consistency check should be made." ) )
+            .withArgument( new OptionalCanonicalPath( "cc-report-dir", "directory", ".",
+                    "Directory where consistency report will be written." ) )
+            .withArgument( new OptionalCanonicalPath( "additional-config", "config-file-path", "",
+                    "Configuration file to supply additional configuration in. This argument is DEPRECATED." ) )
+            .withArgument( new OptionalBooleanArg( "cc-graph", true,
+                    "Perform consistency checks between nodes, relationships, properties, types and tokens." ) )
+            .withArgument( new OptionalBooleanArg( "cc-indexes", true,
+                    "Perform consistency checks on indexes." ) )
+            .withArgument( new OptionalBooleanArg( "cc-label-scan-store", true,
+                    "Perform consistency checks on the label scan store." ) )
+            .withArgument( new OptionalBooleanArg( "cc-property-owners", false,
+                    "Perform additional consistency checks on property ownership. This check is *very* expensive in " +
+                            "time and memory." ) );
```

Comment:
I think the argument names should be extract to constants, so their definition here won't diverge from the code that looks for them to interpret their meaning. The list of constants would also serve as an easier-to-read overview of what options are available.

Reply:
I think it would also be desirable to have an option for overriding the page cache memory setting. People run the backup tool on machines that are already running Neo4j instances, more often than they should.

Reply:
And obviously a page cache size override argument should apply both to the page cache used for the store copy, and the one used for the consistency check.

Reply:
Addressed in next commit